README.md

```diff
@@ -1,19 +1,163 @@
+
+### Storybook.js
+
+Refer [Storybook.js](https://storybook.js.org/) documentation for additional information about stories.
+
+
+### Compile and launch storybook
+
+```bash
+yarn storybook
+```
+
+### Patternlab
+
+```<<<< Placeholder >>>>```
+
+## Component generation
+---
+
+Once theme is generated you can use the theme generator to build components for the theme.
+
+Below is a example of generating story for *button* component,
+
+```bash
+╰─❯ kashmir
+? Please choose what you would like to generate? component
+? Please enter the component name: button
+? Choose the type of the component atom
+? Choose the files you to generate? sass, json, twig, markdown, stories
+The button component was created.
+```
+The button component is already included as an example and can be viewed as part of the storybook out-of-the-box.
+
+### Component structure
+
+The theme genertor would ask you which files you want to generate for the the component,
+- sass - *will contain the styles for component*
+- json - *will contain the data for the component, ex: title, description.*
+- twig - *will contain the markup for the component*
+- markdown - *documentation*
+- stories - *the story js file*
+- typescript - *additional js file*
+
+Based on your component needs, choose which files you need. 
+
+Please check the button component example provided with the theme.
+## Additional Tooling
+---
+
+### RTL suport
+
+We use [postcss-rtlcss](https://www.npmjs.com/package/postcss-rtlcss) for RTL support. You simply right sass as you normally would, the theme compilation will take care of generating compatible RTL CSS.
+
+```bash
+// Input SASS
+
+h2 {
+    text-align: left;
+}
+
+// Generated CSS
+
+[dir="ltr"] h2 {
+  text-align: left;
+}
+
+[dir="rtl"] h2 {
+  text-align: right;
+}
+```
+
+### Image minification
+
+The theme provides inbuild minification via [gulp-imagemin](https://www.npmjs.com/package/gulp-imagemin). This is setup by default, all we have to do is to add the images under `/images` directory.
+
+### TypeScript support
+
+The Kashmir theme comes with support for [TypeScript](https://www.typescriptlang.org/). TypeScript provides all the features of JavaScript with additional features that come with strongly typed programming languages.
+
+The theme provides `main.ts` file as a sample, if you uncomment the typescript code, you could see the generated `main.js` file under `/dist`.
+
+There is no additional step needed, just add a `*.ts` file as and when needed and it shall be compiled by the theme.
+
+### Linting & Prettier support
+
+Linting and prettier support is included, with standard defaults. Please check `.prettierrc`, `.eslintrc` `.eslintignore`.
+
+### Cypress testing support
+
+[Cypress testing framework](https://www.cypress.io/) has been integrated in the theme and a sample test `home.spec.js` is provided.
+
+You can launch cypress from the command line using the below command,
+
+```bash
+yarn cypress run
+```
+
```

Comment:
@skippednote Will add that in as well.

Reply:
@skippednote tailwind css is something I am planning to test, not yet reached there. Both tailwind and patternlab test is pending.

Reply:
@swarad07 I have not completely tested Tailwind. While it worked well with Storybook once we include the commented code, The purge CSS is not working as expected. 

And Pattenlab is blocked with the following issue. 

The issue is how Pattenlab attaches CSS and js in patternlab-config.json. For now, I have supplied a WIP fix part of #245  but the pattenlab serve needs to be discussed and fixed so that the `gulp watch`  proceeds without a fatal error.